pkg/operator/ceph/cluster/watcher.go

```diff
@@ -546,9 +551,54 @@ func (c *clientCluster) createNetworkFence(ctx context.Context, pv corev1.Persis
 	return nil
 }
 
-func (c *clientCluster) unfenceAndDeleteNetworkFence(ctx context.Context, node corev1.Node, cluster *cephv1.CephCluster) error {
+func (c *clientCluster) unfenceAndDeleteRbdNetworkFence(ctx context.Context, node corev1.Node, cluster *cephv1.CephCluster) error {
+	networkFence := &addonsv1alpha1.NetworkFence{}
+	err := c.client.Get(ctx, types.NamespacedName{Name: node.Name + "rbd", Namespace: cluster.Namespace}, networkFence)
```

Comment:
Is there anything specific about an rbd fence or cephfs fence? I assume not since they are both creating the same NetworkFence CR. In that case, the previous approach of having a single NetworkFence CR seems correct. What exactly is the bug we are trying to fix?

Reply:
Issue was here, https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/watcher.go#L519 the nf is created with name as node name. So let say, rbd created 1st network fence, now when cephfs will again come, it will be creating with same name. So, we may fall into https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/watcher.go#L540 and there wont be any errors as well. So we land up on creating nf only once per node.

Reply:
But the expected result would be to have network fence for each driver for whom we have PVs

Reply:
For ref- https://bugzilla.redhat.com/show_bug.cgi?id=2259668

Reply:
Had a discussion with @subhamkrai, I understand now the rbd and cephfs fencing parameters are different, so we need to maintain the different rbd and cephfs networkFence CRs.